3711/1/ruby_gem.rb

```diff
@@ -0,0 +1,43 @@
+# :reek:InstanceVariableAssumption
+# :reek:TooManyConstants
+# :reek:TooManyInstanceVariables
+class RubyGem
+  attr_reader :name, :source_url, :score
+  attr_reader :used_by, :watched_by, :stars, :forks, :contributors, :issues
+
+  WEIGHT_OF_USED_BY       = 10
+  WEIGHT_OF_WATCHED_BY    = 10
+  WEIGHT_OF_STARS         = 10
+  WEIGHT_OF_FORKS         = 10
+  WEIGHT_OF_CONTRIBUTORS  = 10
+  WEIGHT_OF_ISSUES        = 10
+
+  def initialize(gem_data)
+    @gem_date = gem_data
+    fill_attr
+    @score = score_rating
+  end
+
+  private
+
+  def score_rating
+    @used_by * WEIGHT_OF_USED_BY +
+      @watched_by * WEIGHT_OF_WATCHED_BY +
+      @stars * WEIGHT_OF_STARS +
+      @forks * WEIGHT_OF_FORKS +
+      @contributors * WEIGHT_OF_CONTRIBUTORS +
+      @issues * WEIGHT_OF_ISSUES
+  end
+
+  # :reek:TooManyStatements
+  def fill_attr
+    @name         = @gem_date['name']
```

Comment:
This should be refactored in order to fix reek

Reply:
Reworked RubyGem attributes